src/main/scala/com/fulcrumgenomics/FgBioDef.scala

```diff
@@ -41,6 +41,47 @@ class FgBioDef extends CommonsDef {
   implicit class BetterBufferedIteratorJavaWrapper[A](val iterator: java.util.Iterator[A]) {
     def bufferBetter = new BetterBufferedIterator(scala.collection.JavaConversions.asScalaIterator(iterator))
   }
+
+  /**
+    * An implementation of a for loop that has performance similar to writing a custom
+    * while loop for primitive types, which suffer great performance loss when iterating
+    * via foreach(), and especially with zipWithIndex.foreach().
+    *
+    * Equivalent to: for(int i=from; i<until; i+=by) f(i)
+    *
+    * @param from an initial integer values
+    * @param until a value one higher than the last value that should be accepted
+    * @param by a step value to increment by each iteration
+    * @param f a function that takes the index and is called each iteration
+    */
+  @inline def forloop(from: Int, until:Int, by:Int=1)(f: Int => Unit): Unit = {
+    val comp: (Int,Int) => Boolean = if (by > 0) (_ < _) else (_ > _)
```

Comment:
I am curious, is calling this function slower than having an if/else for the two cases?

```
if (by > 0) {
  var i = from 
  while (i < until) {
     ...
}
else {
}